doc/api/esm.md

```diff
@@ -309,6 +309,36 @@ modules it can be used to load ES modules.
 The `import.meta` meta property is an `Object` that contains the following
 properties.
 
+### `import.meta.dirname`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.2 - Release candidate
+
+* {string} The directory name of the current module. This is the same as the
+  [`path.dirname()`][] of the [`import.meta.filename`][].
+
+> **Caveat** only local modules support this property. Modules not using the
+> `file:` protocol will not provide it.
```

Comment:
`local modules` might be misleading. This will break even for modules loaded from `localhost`, which arguably is local, or from in-memory data URLs.

Reply:
Suggestions are welcome. Note, "this will break" is inaccurate. Loading a module from the network, be it remote or local, is not supported by this feature as it is nonsensical.

Reply:
```suggestion
> **Caveat** only file-based modules support this property.
```